### PR TITLE
docs(date-picker): gower review

### PR DESCRIPTION
**Pull request by aagonzales**
Closes https://github.com/carbon-design-system/carbon-website/issues/3045

Recommended immediate updates from @mbgower's review of the date picker docs. Some updates in Mike's original PDF like page structure will be addressed in a future issue. 

[See issue](https://github.com/carbon-design-system/carbon-website/issues/3045#issuecomment-1204289674) for list of changes applied in this PR.

### CODE REVIEW

**Review by dakahn — APPROVED** [review]

File: src/pages/components/date-picker/usage.mdx [hunk 1]
```diff
@@ -30,23 +30,23 @@ times.
...
+Date pickers allow users to select past, present, or future dates. The kind of
+date you are requesting from the user will determine which date picker (simple
+or calendar) is best to use. A time picker helps users select a specific time.
+You can customize the time picker's format depending on location or need.
 
 ### When to use
 
-Use date and time pickers when you are asking the user for a time or date, or
-for scheduling tasks.
+- When asking the user for an exact, approximate, or memorable date or time
```
Comment (laurenmrice) [inline src/pages/components/date-picker/usage.mdx:40]:
```suggestion
- When asking the user for an exact, approximate, or memorable date or time.
```

File: src/pages/components/date-picker/usage.mdx [hunk 2]
```diff
@@ -30,23 +30,23 @@ times.
...
+- For scheduling tasks
```
Comment (laurenmrice) [inline src/pages/components/date-picker/usage.mdx:41]:
```suggestion
- For scheduling tasks.
```

**Review by laurenmrice — CHANGES_REQUESTED** [review]
- On the Style tab under the Structure section for "Time picker", the end of the description sentence has a typo of `Hi` that should be removed.
Screen Shot 2022-08-17 at 10 50 27 AM

- When you show the open calendar menu in images, the previous/next month days should be using `$text-secondary` color token. I think this is a bug in the symbol itself and it just needs to be updated. Might be better to update those images in a later PR once the bug is fixed in the symbol so we don't have to break them all?

Screen Shot 2022-08-17 at 10 54 15 AM

**Comment by aagonzales** [comment]
Oh no, that's a problem with the library component. I'll have to push a change from there first to update the calendars. Good catch though

**Review by tay1orjones — APPROVED** [review]

**Review by laurenmrice — APPROVED** [review]
🎉

**Pull request merged**